### ISSUE TITLE
Unique constraint fails on large text columns

### ISSUE THREAD

**Issue by Nickersoft** [prisma/prisma#15949]
### Bug description

Hey folks, 

I've run into a common Postgres issue using Prisma, but unfortunately, there seems to be no natural way to fix it. I've added a `@@unique()` constraint for two of my database columns: one is a foreign key ID, the other is a `Text` field. Upon inserting/upserting `Text` data with a length over 1,900 characters, I'm receiving:

```
Invalid `prisma.word.upsert()` invocation:

Error occurred during query execution:

ConnectorError(ConnectorError { user_facing_error: None, kind: QueryError(Error { kind: Db, cause: Some(DbError { severity: "ERROR", parsed_severity: Some(Error), code: SqlState(E54000), message: "index row size 3648 exceeds btree version 4 maximum 2704 for index \"documents_text_language_code_key\"", detail: Some("Index row references tuple (0,1) in relation \"documents\"."), hint: Some("Values larger than 1/3 of a buffer page cannot be indexed.Consider a function index of an MD5 hash of the value, or use full text indexing."), position: None, where_: None, schema: Some("public"), table: Some("documents"), column: None, datatype: None, constraint: Some("documents_text_language_code_key"), file: Some("nbtutils.c"), line: Some(2663), routine: Some("_bt_check_third_page") }) }) })
```

**TL;DR:** The error is `Index row references tuple (0,1) in relation "documents". Values larger than 1/3 of a buffer page cannot be indexed.Consider a function index of an MD5 hash of the value, or use full-text indexing.`

This is a common limitation with Postgres UNIQUE fields, and there are several solutions on StackOverflow effectively all saying to just use an md5 hash in a [custom UNIQUE INDEX expression](https://stackoverflow.com/questions/67715283/how-to-insert-long-text-3k-chars-in-columns-with-unique-constraint).

Unfortunately, there doesn't seem to be a good way to do so with Prisma. I tried adding the expression to a migration file, but Prisma can't detect the fact I'm adding a custom UNIQUE index and insists on generating a new migration containing a regular UNIQUE index over the same two columns. I looked at full-text indexing (which only works for MySQL and Mongo, apparently) and full-text search, but neither seems to encapsulate dealing with unique fields, specifically.

I would remove the constraint entirely, except I'm relying on it in the Prisma client to perform upserts using unique tuples. Any guidance on the best approach here would be appreciated.

### How to reproduce

Should be reproducible by adding any `@unique` or `@@unique` constraint to a `String` (`Text`) field in the Prisma schema and trying to insert large data (greater than 1,000 characters). 

### Expected behavior

Ideally, I think Prisma should handle this use case under-the-hood by automatically indexing based on the `md5` function hash. That, or provide some way to hook into full-text indexing for Postgres to prevent this error.

### Prisma information

```prisma
model Document {
  id           BigInt       @id @default(autoincrement())
  text         String
  languageCode LanguageCode @map("language_code") // LanguageCode is an enum
  language     Language     @relation(fields: [languageCode], references: [code])
  @@unique([text, languageCode])
  @@map("documents")
}
```

### Environment & setup

- OS: macOS
- Database: PostgreSQL 14
- Node.js version: 18.9.0

### Prisma Version

```
4.5.0
```

**Comment by janpio** [comment]
This would be covered by https://github.com/prisma/prisma/issues/13414 or https://github.com/prisma/prisma/issues/2504 probably, but for a specific use case?

**Comment by Nickersoft** [comment]
Hmm I suppose it would, as far as using `md5()` is concerned. Are there any imminent plans for this feature to get pushed through? Would it be any easier for the Prisma engine to just account for this specific use case internally? It seems like using `md5` for large text indices is common practice.
